python/dolfinx/io/gmshio.py

```diff
@@ -257,6 +266,20 @@ def model_to_mesh(
         # Sort elements by ascending dimension
         perm_sort = np.argsort(cell_dimensions)
 
+        # Check that all cells are tagged once
+        _d = model.getDimension()
+        assert _d in topologies.keys(), 'All cells are expected to be tagged once; none found'
```

Comment:
As these are user-facing errors they should raise Exceptions of the appropriate type directly, probably RuntimeError, rather than using assert.

Reply:
Done.

Suggestion: could the first RuntimeError (no cell is tagged) be converted into a RuntimeWarning by defaulting a uniform tag for all cells?

Reply:
Currently most warnings get swallowed by FFCx (https://github.com/FEniCS/ffcx/issues/712) so I am not sure this is a good idea